### DISCUSSION TITLE
Issue when running tests after upgrading to 5.2.2

### DISCUSSION THREAD

**Discussion by nikola-lenert** [mikro-orm/nestjs#147] [General]
**Describe the bug**
<img width="1540" alt="image" src="https://github.com/mikro-orm/nestjs/assets/66955219/59b3adab-4d8c-4519-aed8-da5a02bda042">

```
ReferenceError: You are trying to `import` a file after the Jest environment has been torn down. From src/document/subscriptions/document-generator.subscriber.spec.ts.

      at Function.dynamicImport (node_modules/@mikro-orm/core/utils/Utils.js:778:20)
      at Function.getConfiguration (node_modules/@mikro-orm/core/utils/ConfigurationLoader.js:28:52)
      at Function.createEntityManager (node_modules/@mikro-orm/nestjs/mikro-orm-core.module.js:121:32)
      at Function.forRoot (node_modules/@mikro-orm/nestjs/mikro-orm-core.module.js:65:20)
```

Since we upgraded `mikro-orm/nestjs` to version `5.2.2`, the error above occurs 100+ times for various files during the duration of the tests.

**Stack trace**
```
ReferenceError: You are trying to `import` a file after the Jest environment has been torn down. From src/document/subscriptions/document-generator.subscriber.spec.ts.

      at Function.dynamicImport (node_modules/@mikro-orm/core/utils/Utils.js:778:20)
      at Function.getConfiguration (node_modules/@mikro-orm/core/utils/ConfigurationLoader.js:28:52)
      at Function.createEntityManager (node_modules/@mikro-orm/nestjs/mikro-orm-core.module.js:121:32)
      at Function.forRoot (node_modules/@mikro-orm/nestjs/mikro-orm-core.module.js:65:20)
```

**To Reproduce**

Not sure how to reproduce...

**Expected behavior**
The error doesn't happen. 

**Additional context**
Works fine on `mikro-orm/nestjs 5.1.0`.
I tried removing all usage of `MikroOrmModule.forRoot` across tests but it didn't help.
The issue doesn't seem to affect passing of the tests.
Let me know if I can provide any additional info.

**Versions**

| Dependency | Version |
| - | - |
| node | 18 |
| typescript | 5.2.2 |
| mikro-orm | 5.8.4 |
| your-driver | ? |

**Comment by B4nan** [comment]
Sounds like you are missing some await somewhere, the error is quite self explanatory, the `forRoot` happens outside of the test scope/callback. How does your test case look like?

**Reply by B4nan** [reply]
FYI you can see a working setup here, including the [unit tests](https://github.com/mikro-orm/nestjs-realworld-example-app/blob/master/src/tag/tag.controller.spec.ts)

https://github.com/mikro-orm/nestjs-realworld-example-app

**Reply by nikola-lenert** [reply]
Thanks for the quick response.
I scoured the repository for a missing `await` but didn't find anything in the unit tests.

It's hard to pinpoint the problematic file because it seems to throw the error for 70+ different files. 
Also, as you can see, the error stack points to files only found in the package, so we cannot track where the issue starts.
They all initiate the testing modules just like in the example you provided.
For example:
<img width="815" alt="image" src="https://github.com/mikro-orm/nestjs/assets/66955219/a2d8d734-8719-4552-957f-2e851db085db">
<img width="1622" alt="image" src="https://github.com/mikro-orm/nestjs/assets/66955219/d121b56a-78bc-4bde-9b69-65f2e97699ec">

**Reply by B4nan** [reply]
Well, that's where you need to start cutting things unless it starts to work, then you can pinpoint the problematic test case. Or go the other way around, remove everything except one test, if it works, add more till you see it fail. 

This one looks good, but hard to guess based on a screenshot.

**Reply by nikola-lenert** [reply]
Alright, I'll see what can be done and report back on the findings. Thank you! :)

**Reply by B4nan** [reply]
Also be sure to update everything to latest and wipe the lockfile, if it used to work, it might be just a broken installation (but it does not sound like one).
